content/docs/3.0/tasks/try/developers.md

```diff
@@ -0,0 +1,268 @@
+
+For now, please `echo 'nightly-2019-08-25' > ./rust-toolchain` in the project directory.
+{{< /warning >}}
+
+Next, you'll need to add the TiKV client as a dependency in the `Cargo.toml` file:
+
+```toml
+[dependencies]
+tikv-client = { git = "https://github.com/tikv/client-rust.git" }
+tokio = "0.2.0-alpha.4"
+```
+
+Then you can edit the `src/main.rs` file with the following:
+
+```rust
+use tikv_client::{Config, RawClient, Error};
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let config = Config::new(vec!["http://pd.tikv:2379"]);
+    let client = RawClient::new(config)?;
+    let key = "TiKV".as_bytes().to_owned();
+    let value = "Works!".as_bytes().to_owned();
+
+    client.put(key.clone(), value.clone()).await?;
+    println!(
+        "Put: {} => {}",
+        std::str::from_utf8(&key).unwrap(),
+        std::str::from_utf8(&value).unwrap()
+    );
+
+    let returned: Vec<u8> = client.get(key.clone()).await?
+        .expect("Value should be present.").into();
+    assert_eq!(returned, value);
+    println!(
+        "Get: {} => {}",
+        std::str::from_utf8(&key).unwrap(),
+        std::str::from_utf8(&value).unwrap()
+    );
+    Ok(())
+}
+```
+
+{{< info >}}
+TiKV works with binary data to enable your project to store arbitrary data such as binaries or non-UTF-8 encoded data if necessary. While the Rust client accepts `String` values as well as `Vec<u8>`, it will only output `Vec<u8>`.
+{{< /info >}}
+
+Now, because our client needs to be part of the same network (`tikv`) as the PD and TiKV nodes, you must to build this binary into a Docker container. Your `Dockerfile` should look something like this:
+
+```dockerfile
+FROM ubuntu:latest
+# Systemwide setup
+RUN apt update
+RUN apt install --yes build-essential protobuf-compiler curl cmake golang
+
+# Create the non-root user.
+RUN useradd builder -m -b /
+USER builder
+RUN mkdir -p ~/build/src
+
+# Install Rust
+COPY rust-toolchain /builder/build/
+RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain `cat /builder/build/rust-toolchain` -y
+ENV PATH="/builder/.cargo/bin:${PATH}"
+
+# Fetch, then prebuild all deps
+COPY Cargo.toml rust-toolchain /builder/build/
+RUN echo "fn main() {}" > /builder/build/src/main.rs
+WORKDIR /builder/build
+RUN cargo fetch
+RUN cargo build --release
+COPY src /builder/build/src
+RUN rm -rf ./target/release/.fingerprint/tikv-example*
+
+# Actually build the binary
+RUN cargo build --release
+ENTRYPOINT /builder/build/target/release/tikv-example
+```
+
+Next, build the image:
+
+```bash
+docker build -t tikv-example .
+```
+
+Then start the produced image:
+
+```bash
+docker run -ti --rm --network tikv tikv-example
+```
+
+Often it's helpful to combine the build and run steps while doing interactive development. You can use the following command to use the container to build a new binary and run it:
```

Comment:
Seems this build step is taking forever... after I don't know for how long, the build failed with error "error: Could not compile `kvproto`."

Reply:
I am not entirely convinced that we should do such a step that's taking so long for a Get Started guide.

Reply:
That's very strange, it should take no more time than the `docker build` command took. Let me run through the steps again and see if I can replicate it.

Reply:
Let me know if you find the cause.

Reply:
After several attempts to replicate this, I was unable to, but I also decided that the user experience doing this isn't great, so it's better to just forget it :)